extensions.go

```diff
@@ -348,3 +348,114 @@ func (c *Client) SessionAsync() FutureSessionResult {
 func (c *Client) Session() (*btcjson.SessionResult, error) {
 	return c.SessionAsync().Receive()
 }
+
+// FutureSetGenerateResult is a future promise to deliver the result of a
+// AddMiningAddr RPC invocation (or an applicable error).
+type FutureAddMiningAddrResult chan *response
+
+// Receive waits for the response promised by the future and returns an error
+// if any occurred when attempting to add the mining address to the set of
+// addresses.
+func (r FutureAddMiningAddrResult) Receive() error {
+	if _, err := receiveFuture(r); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// AddMiningAddrAsync returns an instance of a type that can be used to get the
+// result of the RPC at some future time by invoking the Receive function on
+// the returned instance.
+//
+// See AddMiningAddr for the blocking version and more details.
+//
+// NOTE: This is a btcsuite extension.
+func (c *Client) AddMiningAddrAsync(addr btcutil.Address) FutureAddMiningAddrResult {
+	cmd := btcjson.NewAddMiningAddrCmd(addr.String())
+	return c.sendCmd(cmd)
+}
+
+// AddMiningAddr attempts to add the passed address to the set of mining
+// addresses for btcd. If the address already exists in the set of mining
+// addresses, then an error is returned.
+//
+// NOTE: This is a btcsuite extension.
+func (c *Client) AddMiningAddr(addr btcutil.Address) error {
+	return c.AddMiningAddrAsync(addr).Receive()
+}
+
+// FutureListMiningAddrsResult is a future promise to deliver the result of a
+// ListMiningAddrs RPC invocation (or an applicable error).
+type FutureListMiningAddrsResult chan *response
+
+// Receive waits for the response promised by the future and returns a list of
+// the current set of mining addresses.
+func (r FutureListMiningAddrsResult) Receive() ([]string, error) {
+	res, err := receiveFuture(r)
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshal result as a list of strings.
+	var result []string
+	err = json.Unmarshal(res, &result)
+	if err != nil {
+		return nil, err
+	}
+
+	return result, nil
+}
+
+// ListMiningAddrsAsync returns an instance of a type that can be used to get the
+// result of the RPC at some future time by invoking the Receive function on
+// the returned instance.
+//
+// See ListMiningAddrs for the blocking version and more details.
+//
+// NOTE: This is a btcsuite extension.
+func (c *Client) ListMiningAddrsAsync() FutureListMiningAddrsResult {
+	cmd := btcjson.NewListMiningAddrsCmd()
+	return c.sendCmd(cmd)
+}
+
+// ListMiningAddrs returns a list of all the currently active mining addresses.
+//
+// NOTE: This is a btcsuite extension.
+func (c *Client) ListMiningAddrs() ([]string, error) {
+	return c.ListMiningAddrsAsync().Receive()
+}
+
+// FutureDelMiningAddrsResult is a future promise to deliver the result of a
+// DelMiningAddr RPC invocation (or an applicable error).
+type FutureDelMiningAddrsResult chan *response
```

Comment:
Along the same lines as the related PRs, would you mind defining all of these delete mining result funcs before the list addresses ones so they're consistently defined in the same order (alphabetical) as btcd/btcutil/btcrpcclient.